rust_src/src/windows.rs

```diff
@@ -153,11 +168,7 @@ pub fn window_margins(window: LispObject) -> LispObject {
             LispObject::constant_nil()
         }
     }
-    let win = if window.is_nil() {
-        selected_window()
-    } else {
-        window
-    }.as_live_window_or_error();
```

Comment:
Note this code was not following what C did. This enforced live window on `selected_window` which the C code simply assumes is true. Some goes for valid checks too.

Reply:
I have also noticed it, but I think something would be seriously broken if this not true.